packages/proto-rpc/src/RpcCommunicator.ts

```diff
@@ -0,0 +1,283 @@
+            const bytes = await this.rpcServerRegistry.onRequest(rpcMessage, callContext)
+            response = this.createResponseRpcMessage({
+                request: rpcMessage,
+                body: bytes
+            })
+        } catch (err) {
+            let responseError = RpcResponseError.SERVER_ERROR
+            if (err.code === ErrorCode.UNKNOWN_RPC_METHOD) {
+                responseError = RpcResponseError.UNKNOWN_RPC_METHOD
+            } else if (err.code === ErrorCode.RPC_TIMEOUT) {
+                responseError = RpcResponseError.SERVER_TIMOUT
+            }
+            response = this.createResponseRpcMessage({
+                request: rpcMessage,
+                responseError
+            })
+        }
+        this.onOutgoingMessage(response, undefined, callContext)
+    }
+
+    private async handleNotification(rpcMessage: RpcMessage, 
+        callContext?: CallContext): Promise<void> {
+        if (this.stopped) {
+            return
+        }
+        try {
+            await this.rpcServerRegistry.onNotification(rpcMessage, callContext)
+        } catch (err) { }
+    }
+
+    private registerRequest(requestId: string, deferredPromises: ResultParts, timeout = this.defaultRpcRequestTimeout): void {
+        if (this.stopped) {
+            return
+        }
+        const ongoingRequest: OngoingRequest = {
+            deferredPromises,
+            timeoutRef: setTimeout(() => this.requestTimeoutFn(deferredPromises), timeout)
+        }
+        this.ongoingRequests.set(requestId, ongoingRequest)
+    }
+
+    private resolveOngoingRequest(response: RpcMessage): void {
+        if (this.stopped) {
+            return
+        }
+        const ongoingRequest = this.ongoingRequests.get(response.requestId)!
+        if (ongoingRequest.timeoutRef) {
+            clearTimeout(ongoingRequest.timeoutRef)
+        }
+        const deferredPromises = ongoingRequest!.deferredPromises
+        this.resolveDeferredPromises(deferredPromises, response)
+        this.ongoingRequests.delete(response.requestId)
+    }
+
+    private rejectOngoingRequest(response: RpcMessage): void {
+        if (this.stopped) {
+            return
+        }
+        const ongoingRequest = this.ongoingRequests.get(response.requestId)!
+        if (ongoingRequest.timeoutRef) {
+            clearTimeout(ongoingRequest.timeoutRef)
+        }
+        let error
+        if (response.responseError === RpcResponseError.SERVER_TIMOUT) {
+            error = new Err.RpcTimeout('Server timed out on request')
+        } else if (response.responseError === RpcResponseError.UNKNOWN_RPC_METHOD) {
+            error = new Err.RpcRequest(`Server does not implement method ${response.header.method}`)
+        } else {
+            error = new Err.RpcRequest('Server error on request')
+        }
+        const deferredPromises = ongoingRequest!.deferredPromises
+        this.rejectDeferredPromises(deferredPromises, error, StatusCodes.SERVER_ERROR)
```

Comment:
```suggestion
        this.rejectDeferredPromises(ongoingRequest!.deferredPromises, error, StatusCodes.SERVER_ERROR)
```

It makes code easier to read if we inline this kind of helper variables which are just aliases to another variables (whenever the alias is very simple). 

That way we can do the code reading linearly:
1. we seem to call a method (`this.rejectDeferredPromises()`)
2. and these are the arguments of that call (`ongoingRequest!.deferredPromises, ...`)

(This code snippet is just an example, the pattern could be applied to multiple places in the codebase.)